h5py/tests/test_group.py

```diff
@@ -877,13 +877,17 @@ def test_copy_group_to_group(self):
     def test_copy_dataset(self):
         self.f1['foo'] = [1,2,3]
         foo = self.f1['foo']
+        grp = self.f1.create_group("grp")
 
         self.f1.copy(foo, 'bar')
         self.assertArrayEqual(self.f1['bar'], np.array([1,2,3]))
 
         self.f1.copy('foo', 'baz')
         self.assertArrayEqual(self.f1['baz'], np.array([1,2,3]))
 
+        self.f1.copy(foo, grp)
+        self.assertArrayEqual(self.f1['/grp/grp'], np.array([1,2,3]))
```

Comment:
The results are not perfect, after copy, the dataset's name remains the same with the group.